### PR TITLE
Update Python package versions

### PR DESCRIPTION
**Pull request by mbengit**
Resolves vulnerabilities in `cryptography` and `werkzeug` packages. Additionally, updates all Python packages to latest versions. 

- This pull request does not affect `qpylib` source code. The updates are for test/build purposes only and do not require the generation of a new `qpylib` release.
- The `pylint` update requires replacing `.pylintrc` content with up-to-date properties.
- The `werkzeug` update requires the modification of one unit test's expected error message text.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-Werkzeug==1.0.1
-requests==2.26.0
-pycryptodome==3.9.6
-cryptography==3.3.2
-responses==0.20.0
-pylint==2.13.3
-pytest==7.1.1
-pytest-cov==3.0.0
-wheel==0.38.1
+Flask==2.2.3
```
Comment (jthomperoo) [inline requirements.txt:1]: The `qradar-app-base` image still uses Flask v1, I wonder should these dependencies mirror the base image?

File: setup.py [hunk 2]
```diff
@@ -25,10 +25,10 @@ def main():
         url="https://github.com/ibm/qpylib",
         packages=setuptools.find_packages(),
         install_requires=[
-            "flask>=1.1,<2",
+            "flask>=1.1",
             "requests>=2.22,<3",
             "pycryptodome>=3.9,<4",
-            "cryptography>=2.8,<4"
+            "cryptography>=3.2"
```
Comment (jthomperoo) [inline setup.py:31]: The `qradar-app-base` v3 stream has cryptography `2.8`, so not sure we can bump this up:
https://ibmsecuritydocs.github.io/qradar_appfw_v2/docs/documentation/qradar_app_base_image_3.0.5_packages.html

File: requirements.txt (hunk 1 above)
Comment (mbengit) [reply requirements.txt:1]: Does that matter? Realistically, these requirements are only for build/test, not for base image install and execution. If we stick with older package versions in this project then we're going to hit more Dependabot issues, e.g. cryptography was 3.3.2 and we need 39.0.1 to fix a critical vulnerability.

File: setup.py (hunk 2 above)
Comment (mbengit) [reply setup.py:31]: Good spot, thanks. I was using the v2 image's SBOM, forgot to check v3.

File: requirements.txt (hunk 1 above)
Comment (jthomperoo) [reply requirements.txt:1]: Yep good point, these dependencies are only for the build pipeline so don't need to match.
Looks good!

**Review by jthomperoo — APPROVED** [review]

**Pull request merged**